sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,841 @@
+                    continue;
+                }
+
+                IParameterSymbol? parameterSymbol = model.GetDeclaredSymbol(parameter);
+
+                if (parameterSymbol is null)
+                {
+                    throw new InvalidOperationException($"The symbol for the parameter '{nameof(parameter)}' could not be found");
+                }
+
+                // Check to see if any of the attributes associated with this parameter is a BindingAttribute
+                foreach (var attribute in parameterSymbol.GetAttributes())
+                {
+                    if (IsBindingAttribute(attribute))
+                    {
+                        string? dataType = null;
+                        string parameterSymbolType = parameterSymbol.Type.GetFullName();
+
+                        if (IsHttpTrigger(attribute))
+                        {
+                            hasHttpTrigger = true;
+                        }
+                        else if (IsEventHubsTrigger(attribute))
+                        {
+                            if (!IsEventHubsTriggerValid(parameterSymbol, attribute, out dataType)) // we need the parameterSymbol to validate the EventHubs trigger, so we'll validate it at this step
+                            {
+                                // report diagnostic error
+                            }
+                        }
+
+                        if (dataType is null)
+                        {
+                            dataType = GetDataTypeFromType(parameterSymbolType);
+                        }
+
+                        string bindingName = parameter.Identifier.ValueText;
+
+                        WriteBindingToFile(indentedTextWriter, attribute, funcName, bindingName, dataType);
+                    }
+                }
+            }
+        }
+
+
+        /// <summary>
+        /// Adds bindings found in the ReturnType class of the Function
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Throws when a symbol cannot be found.</exception>
+        /// <exception cref="FormatException">Throws if the attributes were used and formatted incorrectly.</exception>
+        private static void AddReturnTypeBindings(IndentedTextWriter indentedTextWriter, MethodDeclarationSyntax method, SemanticModel model, string funcName, bool hasHttpTrigger, bool hasOutputBinding)
+        {
+            TypeSyntax returnTypeSyntax = method.ReturnType;
+            ITypeSymbol? returnTypeSymbol = model.GetSymbolInfo(returnTypeSyntax).Symbol as ITypeSymbol;
+
+            if (returnTypeSymbol is null)
+            {
+                throw new FormatException("Symbol can't be found.");
+            }
+
+            if (returnTypeSymbol != null &&
+                !String.Equals(returnTypeSymbol.GetFullName(), Constants.VoidType, StringComparison.Ordinal) &&
+                !String.Equals(returnTypeSymbol.GetFullName(), Constants.TaskType, StringComparison.Ordinal))
```

Comment:
Is this the recommended way for checking types? What do you think of having access to the `compilation` object here and inverting this: `typeSymbol.Equals(returnTypeSymbol.GetTypeByMetadataName("System.Threading.Tasks.Task"))`